src/spikeinterface/core/sorting_tools.py

```diff
@@ -220,3 +220,102 @@ def random_spikes_selection(
         raise ValueError(f"random_spikes_selection(): method must be 'all' or 'uniform'")
 
     return random_spikes_indices
+
+
+def get_ids_after_merging(sorting, units_to_merge, new_unit_ids=None):
+    merged_unit_ids = set(sorting.unit_ids)
+    for count in range(len(units_to_merge)):
+        assert len(units_to_merge[count]) > 1, "A merge should have at least two units"
+        for unit_id in units_to_merge[count]:
+            assert unit_id in sorting.unit_ids, "Merged ids should be in the sorting"
+        if new_unit_ids is None:
+            for unit_id in units_to_merge[count][1:]:
+                merged_unit_ids.discard(unit_id)
+        else:
+            for unit_id in units_to_merge[count]:
+                merged_unit_ids.discard(unit_id)
+            merged_unit_ids = merged_unit_ids.union([new_unit_ids[count]])
```

Comment:
I think I prefer to give a new unit id to the merge, rather than keeping the first entry:

You can port [this function](https://github.com/SpikeInterface/spikeinterface/blob/main/src/spikeinterface/curation/mergeunitssorting.py#L52-L74) to `sorting_tools`. In case unit ids are strings, the new unit will be `unit1-unit2-...`

This is another advocate IMO to force unit ids to be strings....